### ISSUE TITLE
[turborepo] dependsOn does not wait for all ^ tasks to complete before starting tasks without ^ prefix

### ISSUE THREAD

**Issue by gajus** [vercel/turbo#3314]
### What version of Turbopack are you using?

1.7.0

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

In our case, the project build hierarchy is that every project depends on its parent being first `generate`(d) followed by `build`.

Logically, this should be:

```json
"build": {
  "dependsOn": [
    "^generate",
    "^build",
    "generate"
  ],
  "outputs": [
    "dist/**"
  ]
},
```

However, it seems that the `generate` task (#3 entry) starts before all of the `^generate` _and_ `^build` tasks complete.

And because `generate` (#3 entry) depends on all generate and build of parents completing first, it fails.

In short, there appears to be no way to say:

1. first `generate` all parents workspace
2. then `build` all parents workspace
3. then `generate` current workspace
4. then `build` current workspace

at least not in one command.

Using multiple commands, I am able to do it:

```
pnpm turbo generate --filter=foo^...
pnpm turbo build --filter=foo^...
pnpm turbo generate --filter=foo
pnpm turbo build --filter=foo
```

However, this approach requires that we do _not_ define any dependencies in `turbo.json`.

### Expected Behavior

Expecting all dependencies to first be generated, then built, then the current workspace to be generated and then built.

### To Reproduce

N/A

**Comment by gajus** [comment]
Turns out it was a false-positive.

This may help others https://www.loom.com/share/e94695a2c9fe417ea9271bf48fa0d832
